File: RockPaperScissors.java
//******************************************************************************************
//RockPaperScissors.java		 	Author: Harrison Fox 
//
//Implements a class to represent and play a game of rock, paper and scissors.
//
//******************************************************************************************
import java.util.*;

// public class RockPaperScissors implements Comparable{
public class RockPaperScissors{
	private RockPaperScissorsHand userHand = new RockPaperScissorsHand();
	private RockPaperScissorsHand computerHand = new RockPaperScissorsHand();
	int gameResult;
	String outcome;

	public void play(int userChoice){
		int userHandChoice = userChoice;
		computerHand.computerChoice();
		userHand.setHand(userHandChoice);

		gameResult = computerHand.compareTo(userHand);
		
		if (gameResult == 0){
			outcome = "it's a tie!";
		}
		if (gameResult > 0){
			outcome = "the computer wins!";
		}
		if (gameResult < 0){
			outcome = "you win!";
		}
		
	}
	public String toString(){

		String gameOutcome = "You chose " + userHand + " the computer chose " + computerHand + " and " + outcome;
		
		return gameOutcome;

	}

}

File: RockPaperScissors.js
var userChoice = prompt("Do you choose rock, paper or scissors?");
if (userChoice === "rock" || userChoice === "paper" || userChoice === "scissors"){
    console.log("Get ready to play!");
}
else{
    userChoice = prompt("You did not enter one of the three possible entries, please try again.");
}

var computerChoice = Math.random();
if (computerChoice < 0.34) {
	computerChoice = "rock";
} else if(computerChoice <= 0.67) {
	computerChoice = "paper";
} else {
	computerChoice = "scissors";
} console.log("Computer: " + computerChoice);

var compare = function (choice1, choice2){
    if (choice1 === choice2){
        userChoice = prompt("The result is a tie! Try again!");
        computerChoice = Math.random();
        if (computerChoice < 0.34) {
	        computerChoice = "rock";
        } else if(computerChoice <= 0.67) {
	        computerChoice = "paper";
        } else {
	        computerChoice = "scissors";
        } console.log("Computer: " + computerChoice);
         compare(userChoice, computerChoice);
    }
    else if (choice1 === "rock") {
        if (choice2 === "scissors"){
            return "rock wins!";
        }
        else{
            return "paper wins!";
        }
    }
    else if (choice1 === "paper"){
        if (choice2 === "rock"){
            return "Paper wins!";
        }
        else{
            return "scissors wins!";
        }
    }
    else if (choice1 === "scissors"){
        if (choice2 === "rock"){
            return "rock wins!";
        }
        else{
            return "scissors wins!";
        }
    }
};

compare(userChoice, computerChoice);

